javaslang/src/test/java/javaslang/collection/TreeMapTest.java

```diff
@@ -121,10 +127,312 @@ public void shouldWrapMap() {
+        assertThat(unit.headMap("nancy", false)).isEqualTo(TreeMap.ofEntries(aaron, carter, john));
+        assertThat(unit.headMap("nancy")).isEqualTo(TreeMap.ofEntries(aaron, carter, john));
+    }
+
+    // TailMap
+
+    @Test
+    public void canCreateTailMap() {
+        final TreeMap<String, Integer> unit = TreeMap.ofEntries(aaron, carter, john, nancy, simon);
+        assertThat(unit.tailMap("carter", true)).isEqualTo(TreeMap.ofEntries(carter, john, nancy, simon));
+        assertThat(unit.tailMap("carter", false)).isEqualTo(TreeMap.ofEntries(john, nancy, simon));
+        assertThat(unit.tailMap("carter")).isEqualTo(TreeMap.ofEntries(carter, john, nancy, simon));
+    }
+
+    // Floor
+
+    @Test
+    public void canFloor() {
+        final TreeMap<String, Integer> unit = TreeMap.ofEntries(aaron, carter, john, nancy, simon);
+        assertThat(unit.floor("john")).isEqualTo(Option.of(john));
+        assertThat(unit.floor("eddie")).isEqualTo(Option.of(carter));
+        assertThat(unit.floor("aaa")).isEqualTo(Option.none());
+    }
+
+    @Test
+    public void canFloorKey() {
+        final TreeMap<String, Integer> unit = TreeMap.ofEntries(aaron, carter, john, nancy, simon);
+        assertThat(unit.floorKey("john")).isEqualTo(Option.of("john"));
+        assertThat(unit.floorKey("eddie")).isEqualTo(Option.of("carter"));
+        assertThat(unit.floor("aaa")).isEqualTo(Option.none());
+    }
+
+    // Ceiling
+
+    @Test
+    public void canCeiling() {
+        final TreeMap<String, Integer> unit = TreeMap.ofEntries(aaron, carter, john, nancy, simon);
+        assertThat(unit.ceiling("john")).isEqualTo(Option.of(john));
+        assertThat(unit.ceiling("eddie")).isEqualTo(Option.of(john));
+        assertThat(unit.ceiling("xxx")).isEqualTo(Option.none());
+    }
+
+    @Test
+    public void canCeilingKey() {
+        final TreeMap<String, Integer> unit = TreeMap.ofEntries(aaron, carter, john, nancy, simon);
+        assertThat(unit.ceilingKey("john")).isEqualTo(Option.of("john"));
+        assertThat(unit.ceilingKey("eddie")).isEqualTo(Option.of("john"));
+        assertThat(unit.ceilingKey("xxx")).isEqualTo(Option.none());
+    }
+
+    // Lower
+
+    @Test
+    public void canLower() {
+        final TreeMap<String, Integer> unit = TreeMap.ofEntries(aaron, carter, john, nancy, simon);
+        assertThat(unit.lower("john")).isEqualTo(Option.of(carter));
+        assertThat(unit.lower("eddie")).isEqualTo(Option.of(carter));
+        assertThat(unit.lower("aaa")).isEqualTo(Option.none());
+    }
+
+    @Test
+    public void canLowerKey() {
+        final TreeMap<String, Integer> unit = TreeMap.ofEntries(aaron, carter, john, nancy, simon);
+        assertThat(unit.lowerKey("john")).isEqualTo(Option.of("carter"));
+        assertThat(unit.lowerKey("eddie")).isEqualTo(Option.of("carter"));
+        assertThat(unit.lowerKey("aaa")).isEqualTo(Option.none());
```

Comment:
> Haven't you seen Coach Carter? ;)

Examples should be cross-cultural, e.g. numbers, like the ones you provided, but with better naming (i.e. not the same as the content, but representing **why** it was chosen or **how** it will be used, i.e. its intention, not its content (:).

e.g. `minimumValue`, `smallValue`, `bigvalue`, `maximumValue` or something like that :)